todo/src/components/TodoList.js

```diff
@@ -1,34 +1,105 @@
 import React, { useState } from "react"
 import TodoItem from "./TodoItem"
 import uuid from "react-uuid"
+import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd"
+import "../App.css"
 
 function TodoList(props) {
 	const [task, setTask] = useState("Add a task")
 	const [todos, setTodos] = useState(props.todos)
+	const [counter, setCounter] = useState(0)
+	const [filter, setFilter] = useState(" All")
+
+	const onDragEnd = (result) => {
+		const { destination, source } = result
+
+		if (!destination) {
+			return
+		}
+
+		if (destination.index === source.index) {
+			return
+		}
+
+		const newTasks = Array.from(todos)
+		const [removed] = newTasks.splice(source.index, 1)
+		newTasks.splice(destination.index, 0, removed)
+		setTodos(newTasks)
+	}
 
 	const handleChange = (event) => {
 		setTask(event.target.value)
 	}
 
 	const handleSubmit = (event) => {
 		event.preventDefault()
-		const newTodo = { content: task, isComplete: false }
-		setTodos([...todos, newTodo])
-		setTask("Add a task")
+		if (!(task === "Add a task" || task === "")) {
+			const newTodo = { content: task, isComplete: false, id: uuid() }
+			setTodos([...todos, newTodo])
+			setCounter(counter + 1)
+		}
+		setTask("")
 	}
+	const deleteTask = (id) => {
+		const index = todos.findIndex((item) => item.id === id)
 
+		if (index !== -1) {
+			todos.splice(index, 1)
+		}
+		setTodos([...todos])
+		setCounter(counter - 1)
+	}
+	const clearCompleted = () => {
+		const completedTasks = todos.filter((task) => !task.isComplete)
+		setTodos(completedTasks)
+		setCounter(completedTasks.length)
+	}
+	const checkBoxChange = (id) => {
+		const index = todos.findIndex((item) => item.id === id)
+		todos[index].isComplete = !todos[index].isComplete
+		setTodos([...todos])
+	}
+	const showKindsOfLists = (event) => {
+		const val = event.target.innerHTML
+		if (val === " All") {
+			setFilter(val)
+		} else if (val === " Active") {
+			setFilter(val)
+		} else if (val === " Completed") {
+			setFilter(val)
+		}
+	}
+	let filteredTasks = todos
+	if (filter === " Active") {
+		filteredTasks = todos.filter((task) => !task.isComplete)
+	} else if (filter === " Completed") {
+		filteredTasks = todos.filter((task) => task.isComplete)
+	}
 	return (
-		//hi
-		<div style={{ width: "55vw" }}>
+		<div
+			style={{
+				width: "55vw",
+			}}
+		>
 			<div
 				style={{
 					marginBottom: "10px",
+					minWidth: "100%",
 				}}
 			>
 				<form onSubmit={handleSubmit}>
-					<input type="submit" value="+" />
+					<button id="plus" sytle={{}} type="submit" value="+">
+						+
+					</button>
 					<label>
 						<input
+							style={{
+								minWidth: "95%",
+								display: "relaive",
+								backgroundColor: props.darkModeEnabled ? "#333333" : "#f1f1f1",
+								color: props.darkModeEnabled ? "#f1f1f1" : "#333333",
+								borderRadius: "10px",
+							}}
 							type="text"
 							value={task}
 							onClick={() => setTask("")}

```

Comment:
Why would you set the task to empty string every time you click on the input? I wouldn't want the task I wrote to be deleted if I pressed on the input before adding it.